### ISSUE TITLE
[BUG] Generator crashes if openapi-spec contains a circular reference

### ISSUE THREAD

**Issue by JustSamuel** [enzonotario/vitepress-theme-openapi#57]
Take the following minimal crashable example:

```json
{
  "openapi": "3.0.0",
  "info": {
    "title": "Minimal API with Circular Reference",
    "version": "1.0.0"
  },
  "paths": {
    "/parent": {
      "get": {
        "summary": "Get a parent",
        "operationId": "getParent",
        "responses": {
          "200": {
            "description": "A parent with a child",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Parent"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "Parent": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string"
          },
          "child": {
            "$ref": "#/components/schemas/Child"
          }
        }
      },
      "Child": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string"
          },
          "parent": {
            "$ref": "#/components/schemas/Parent"
          }
        }
      }
    }
  }
}
```

This is valid openapi spec as can be seen checked using a [validator](https://apitools.dev/swagger-parser/online/). Of course, dealing with a circular swagger requires a form of normalization. 

I am currently using the standard swagger UI, which does nicely deal with circular references. However, as I have moved all my documentation over to vitepress, I would also prefer to maybe include some swagger here and there as it could help make explanations a bit clearer.

**Comment by enzonotario** [comment]
I fixed it in #60 . Can you please validate it?

**Comment by JustSamuel** [comment]
Works like a charm!
